docs/build/spells.md

```diff
@@ -0,0 +1,128 @@
+
+Think of them as of automated jobs, that can be executed when some specific trigger event occurs (time, connection poll, http, blockchain events, etc).
+
+:::info
+Older implementation of Fluence had Scheduled Scripts, stateless AIR scripts which were periodically run by the node. There had rather restricted use cases since:
+- they were stateless
+- they could be triggered only by a timer
+- they were working on behalf of the node
+- they couldn’t communicate with each other
+- they didn’t have a good management interface
+
+Spells are basically Scheduled Scripts done ~~right~~ better.
+:::
+
+Spells can be used for user-defined automation scenarios for service maintenance, periodical data extraction, subnet maintenance, etc. 
+
+A spell is a marine service, the main goal of which is to provide state storage for script execution. It also provides some additional features to ease the spell usage for users and take some responsibility from the node (aka Nox). The service serves as a template and is instantiated on the creation of a new spell.
+
+This service is intended to be used as a template and instantiated by a node on each new spell creation.
+
+## Managing spells
+Nox provides user API for interacting with spells.
+
+Users are able to control their spells' lifecycle: install new ones, update them, and remove them. The developer experience is provided by the means of Fluence CLI. 
+
+Installing spells include providing the following: 
+- The spell script, which will be executed
+- The list of trigger events (aka the trigger configuration) on which to run the spell
+- The initial state of the spell (e.g. initial state of spell's KV)
+
+After installation, the developer is able to update the initial settings: the script, the trigger configuration, and the state KV storage.
+
+Also, the developer is able to remove their spells, cleaning the state from the peer.
+
+Users can check their spells' state: 
+- inspect their internal state (the KV storage, the script, the trigger config)
+- check if they’re currently subscribed to some triggers
+- obtain their peer identity: spell_id and worker_id.
+
+## Events that can trigger spells
+Nox triggers spells according to their configuration. Currently Nox provides the following event triggers:
+- Timer/CRON Triggers: run spells at a specified time or within a specified period.
+Users should be able to specify the following:
+    - when to run the spell,
+    - when to unsubscribe the spell from timer events,
+    - with what period to run the spell.
+- Connection Triggers: run spells on connection/disconnection of some peers to the peer.
+- All other trigger types can be implemented based on the timer triggers & the spell KV storage (blockchain event trigger, database event trigger, HTTP event trigger, etc)
+
+## Creating and updating spells
+
+Spells can be created and managed with Fluence CLI. 
+
+Let's create a new spell in a recently scaffolded Fluence CLI project by running `fluence spell new`:
```

Comment:
It'd be nice to place here a link to other doc page where we create a quickstart project and explain the basics